packages/data/src/redux-store/index.js

```diff
@@ -148,6 +151,14 @@ export default function createReduxStore( key, options ) {
 				},
 				store
 			);
+			lock(
+				actions,
+				new Proxy( privateActions, {
+					get: ( target, prop ) => {
+						return mapActions( privateActions, store )[ prop ];
```

Comment:
Would be great to only call `mapActions()` once and not on each call. Ditto for `mapSelectors()` below

Reply:
I'll refrain from optimizing this prematurely – `mapActions` is cheap and lazy mapping requires reconciling the already mapped `privateActions` and the current `privateActions`.

Reply:
Would `memo()` around the `get()` be an easy way to do this?

Reply:
Oh, hm, yes it would! Perhaps something like:

```js
import memoize from 'memize';

// ...

get: memoize( (target, prop) => {
```

It would require crafting a few smart test cases for different before/after scenarios so I won't block this PR on it. If you'd like to play with it and spin a new PR, I'll gladly help with the reviews!